File: programmers/lv0/코딩기초트레이닝/조건문/원소들의곱과합/Solution.go
package main

import "fmt"

func solution(num_list []int) int {
	sum := 0
	mul := 1
	for _, x := range num_list {
		sum += x
		mul *= x
	}
	if mul < sum*sum {
		return 1
	}
	return 0
}

func main() {
	fmt.Println(solution([]int{3, 4, 5, 2, 1}))
	fmt.Println(solution([]int{5, 7, 8, 3}))
}

File: programmers/lv0/코딩기초트레이닝/조건문/원소들의곱과합/Solution.java
import java.util.stream.IntStream;

class Solution {
    public static int solution(int[] num_list) {
        int mul = IntStream.of(num_list).reduce(1, (l, r) -> l * r);
        int sum = IntStream.of(num_list).reduce(0, Integer::sum);
        return mul < sum * sum ? 1 : 0;
    }

    public static void main(String[] args) {
        System.out.println(solution(new int[]{3, 4, 5, 2, 1}));
        System.out.println(solution(new int[]{5, 7, 8, 3}));
    }
}